src/components/Dialogs/Conditions/NumericCondition.js

```diff
@@ -0,0 +1,87 @@
+import i18n from '@dhis2/d2-i18n'
+import { SingleSelectField, SingleSelectOption, Button, Input } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React from 'react'
+import classes from './styles/Condition.module.css'
+
+const NULL_VALUE = 'NV'
+export const OPERATOR_EQUAL = 'EQ'
+export const OPERATOR_GREATER = 'GT'
+export const OPERATOR_GREATER_OR_EQUAL = 'GE'
+export const OPERATOR_LESS = 'LT'
+export const OPERATOR_LESS_OR_EQUAL = 'LE'
+export const OPERATOR_NOT_EQUAL = '!EQ'
+export const OPERATOR_EMPTY = `EQ:${NULL_VALUE}`
+export const OPERATOR_NOT_EMPTY = `NE:${NULL_VALUE}`
+
+const operators = {
+    [OPERATOR_EQUAL]: 'equal to (=)',
+    [OPERATOR_GREATER]: 'greater than (>)',
+    [OPERATOR_GREATER_OR_EQUAL]: 'greater than or equal to (≥)',
+    [OPERATOR_LESS]: 'less than (<)',
+    [OPERATOR_LESS_OR_EQUAL]: 'less than or equal to (≤)',
+    [OPERATOR_NOT_EQUAL]: 'not equal to (≠)',
+    [OPERATOR_EMPTY]: 'is empty / null',
+    [OPERATOR_NOT_EMPTY]: 'is not empty / not null',
+}
+
+const NumericCondition = ({ condition, onChange, onRemove }) => {
+    const setOperator = input => {
+        if (input.includes(NULL_VALUE)) {
+            onChange(`${input}`)
+        } else {
+            onChange(`${input}:${value || ''}`)
+        }
+    }
```

Comment:
Should this function accept two args: `input` and `value`?
I see `value` is being used in the `else` clause but I don't see where its value would come from.

Reply:
`operator` and `value` are global consts for the `NumericCondition` and are set here https://github.com/dhis2/event-reports-app/blob/67d6a29aca8d2037e68f39105d5d67a451485e7d/src/components/Dialogs/Conditions/NumericCondition.js#L38
Maybe this should be renamed or the whole section be reordered to above this fn?

Reply:
I had to think a for a minute why this code actually runs, because the functions that use these variables are declared before these variables themselves. So yeah, I think it would help (me) if you moved this variable declaration to above line 29.